### PR TITLE
[WIP] [UI Tests] Minor refactoring around "e2eAllDayStatsLoad" UI test.

### PR DESCRIPTION
**Pull request by pachlava**
### What
There was an influx of fails of `e2eAllDayStatsLoad` for JP recently, and before realizing that they are coming from [release/21.9.1](https://github.com/wordpress-mobile/WordPress-Android/tree/release/21.9.1) branch, which did not contain the fix (#18074) yet, I tried to replicate the fail locally. For some reason, it started failing in a different place, while trying to tap `Stats` button. So, I've adjusted a locator a bit, which solved the local issue, and at the same time seems not to break the CI execution.

### To test
- See that the CI is green for both WP and JP
- I also have a different sort-of "test" PR #18152, where there are more executions, none of them failed.

### CODE REVIEW

File: WordPress/src/androidTest/java/org/wordpress/android/support/WPSupportUtils.java [hunk 1]
```diff
@@ -818,6 +819,10 @@ public static void scrollIntoView(Integer scrollableContainerID, ViewInteraction
     }
 
     public static void dismissJetpackAdIfPresent() {
+        if (BuildConfig.IS_JETPACK_APP) {
+            return;
+        }
+
```
Comment (pachlava) [inline WordPress/src/androidTest/java/org/wordpress/android/support/WPSupportUtils.java:825]: We don't expect a JP banner in JP app, so we can return from this function if we're using JP.

File: WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/MySitesPage.kt [hunk 2]
```diff
@@ -154,7 +154,11 @@ class MySitesPage {
 
     fun goToStats(): StatsPage {
         goToMenuTab()
-        clickQuickActionOrSiteMenuItem(R.id.quick_action_stats_button, R.string.stats)
+        val statsButton = Espresso.onView(Matchers.allOf(
+            ViewMatchers.withText(R.string.stats),
+            ViewMatchers.withId(R.id.my_site_item_primary_text)
+        ))
+        WPSupportUtils.clickOn(statsButton)
```
Comment (pachlava) [inline WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/MySitesPage.kt:161]: Instead of trying to tap the `Stats` button using one of the two alternative locators (one for `Stats` button in `Quick Links`, and another one in the classical` Menu`) I went for using one in classical menu, which is always there.

**Review by ParaskP7 — APPROVED** [review]
👋 @pachlava !

I have reviewed and tested this PR as per the instructions, everything works as expected, thanks for taking care of that! 🌟 🌟 🌟
